OpenRA.Mods.Common/Projectiles/LaserZap.cs

```diff
@@ -112,10 +132,20 @@ public LaserZap(LaserZapInfo info, ProjectileArgs args, Color color)
 
 			if (!string.IsNullOrEmpty(info.HitAnim))
 				hitanim = new Animation(args.SourceActor.World, info.HitAnim);
+			else
+				animationComplete = true;
+
+			hasLaunchEffect = !string.IsNullOrEmpty(info.LaunchEffectImage) && !string.IsNullOrEmpty(info.LaunchEffectSequence);
 		}
 
 		public void Tick(World world)
 		{
+			source = args.CurrentSource();
+
+			if (hasLaunchEffect && ticks < info.LaunchEffectDuration)
```

Comment:
If `info.LaunchEffectDuration` > 1 then this will add a new effect every tick.  I could understand this if the interval were configurable, but do we really have a use case for every tick?

Reply:
Well, I do in my mod, but I guess you're right that making the interval configurable would be better.